src/overlays/actors/ovl_En_Wood02/z_en_wood02.c

```diff
@@ -29,15 +60,392 @@ const ActorInit En_Wood02_InitVars = {
+
+extern Gfx D_06000700[];
+
+s32 func_80B3AF70(EnWood02* this, GlobalContext* globalCtx, Vec3f* pos) {
+    f32 phi_f12;
+
+    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->mf_11D60, pos, &this->actor.projectedPos, &this->actor.projectedW);
+
+    phi_f12 = ((this->actor.projectedW == 0.0f) ? 1000.0f : fabsf(1.0f / this->actor.projectedW));
+
+    if ((-this->actor.uncullZoneScale < this->actor.projectedPos.z) &&
+        (this->actor.projectedPos.z < (this->actor.uncullZoneForward + this->actor.uncullZoneScale)) &&
+        (((fabsf(this->actor.projectedPos.x) - this->actor.uncullZoneScale) * phi_f12) < 1.0f) &&
+        (((this->actor.projectedPos.y + this->actor.uncullZoneDownward) * phi_f12) > -1.0f) &&
+        (((this->actor.projectedPos.y - this->actor.uncullZoneScale) * phi_f12) < 1.0f)) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnWood02_SpawnOffspring(EnWood02* this, GlobalContext* globalCtx) {
+    EnWood02* childWood;
+    s16* childSpawnAngle;
+    Vec3f childPos;
+    s16 extraRot;
+    s16 childParams;
+    s32 i;
+
+    for (i = 4; i >= 0; i--) {
+        if ((this->unk_14E[i] & 0x7F) == 0) {
+            extraRot = 0;
+            if (this->actor.params == BUSH_GREEN_LARGE_SPAWNER) {
+                extraRot = 0x4000;
+            }
+            childSpawnAngle = &sSpawnAngle[i];
+            sSpawnCos = Math_CosS(*childSpawnAngle + this->actor.posRot.rot.y + extraRot);
+            sSpawnSin = Math_SinS(*childSpawnAngle + this->actor.posRot.rot.y + extraRot);
+            childPos.x = (sSpawnDistance[i] * sSpawnSin) + this->actor.initPosRot.pos.x;
+            childPos.y = this->actor.initPosRot.pos.y;
+            childPos.z = (sSpawnDistance[i] * sSpawnCos) + this->actor.initPosRot.pos.z;
+            if (func_80B3AF70(this, globalCtx, &childPos)) {
+                if ((this->unk_14E[i] & 0x80) != 0) {
+                    childParams = (0xFF00 | (this->actor.params + 1));
+                } else {
+                    childParams = (((this->drawType & 0xF0) << 4) | (this->actor.params + 1));
+                }
+                childWood = (EnWood02*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx,
+                                                          ACTOR_EN_WOOD02, childPos.x, childPos.y, childPos.z,
+                                                          this->actor.posRot.rot.x, *childSpawnAngle, 0, childParams);
+                if (childWood != NULL) {
+                    childWood->unk_14E[0] = i;
+                    this->unk_14E[i] |= 1;
+                    childWood->actor.projectedPos = this->actor.projectedPos;
+                } else {
+                    this->unk_14E[i] &= 0x80;
+                }
+            }
+        }
+    }
+}
+
+void EnWood02_Init(Actor* thisx, GlobalContext* globalCtx2) {
+
+    s16 spawnType;
+    f32 actorScale;
+    GlobalContext* globalCtx = globalCtx2;
+    EnWood02* this = THIS;
+    CollisionPoly* outPoly;
+    void* outBgId;
```

Comment:
the type for bgid can prob be updated according to the new bgcheck code that just got merged